pkg/controller/usersignup/usersignup_controller.go

```diff
@@ -46,6 +46,40 @@ const (
 
 var log = logf.Log.WithName("controller_usersignup")
 
+type BannedUserToUserSignupMapper struct {
+	client client.Client
+}
+
+func (b BannedUserToUserSignupMapper) Map(obj handler.MapObject) []reconcile.Request {
+	if bu, ok := obj.Object.(*toolchainv1alpha1.BannedUser); ok {
+		// look-up any associated UserSignup using the BannedUser's "toolchain.dev.openshift.com/emailHash" label
+		if emailHashLbl, exists := bu.Labels[toolchainv1alpha1.BannedUserEmailHashLabelKey]; exists {
+
+			labels := map[string]string{toolchainv1alpha1.BannedUserEmailHashLabelKey: emailHashLbl}
+			opts := client.MatchingLabels(labels)
+			userSignupList := &toolchainv1alpha1.UserSignupList{}
+			if err := b.client.List(context.TODO(), userSignupList, opts); err != nil {
+				// What should we do here??
```

Comment:
I believe we can't do much except logging the error...

Reply:
I've added a log statement here.